cmd/snap-update-ns/utils.go

```diff
@@ -301,7 +301,7 @@ func secureMklinkAll(name string, perm os.FileMode, uid sys.UserID, gid sys.Grou
 		return err
 	}
 	// TODO: roll this uber securely like the code above does using linkat(2).
-	err = osSymlink(oldname, name)
+	err = sysSymlink(oldname, name)
 	if err == syscall.EROFS {
```

Comment:
Come to think of it, the check should probably be:
```
if errno := err.(syscall.Errno) && errno == syscall.EROFS {
   ....
}
```
Note, that it will intentionally panic if the error is not a syscall.Errno as we need to have exactly that type.

Reply:
We have a few more places that check things like that. I don't know if we should panic (I really want this to be a compile error) but .. perhaps? I can follow up of there's consensus on this.